### ISSUE TITLE
Allow Input With Blur Change

### ISSUE THREAD

**Issue by kmccullough** [shipshapecode/ember-flatpickr#416]
When the user types a date into the input box, and then blurs to the next field, nothing is committed to the onChange action and the calendar is not updated; only after pressing enter in the field. Is there a way to configure ember-flatpickr to onChange whenever the date in the input box becomes a valid date, or at least specifically on blur?

**Comment by rwwagner90** [comment]
@kmccullough this is a thin wrapper around flatpickr, so we pretty much only support what flatpickr supports. I am not sure if it is supported in vanilla flatpickr or not.

**Comment by kmccullough** [comment]
If you cannot implement a fix for this, could you recommend a method for safely accessing the flatpickr input control, from the component using the ember-flatpickr component, such that this functionality may be added?

**Comment by rwwagner90** [comment]
@kmccullough I was saying in my previous comment that I don't know if flatpickr supports this or not. Is there something in the flatpickr docs that you see that makes you think it is supported?

**Comment by kmccullough** [comment]
I am not familiar with flatpickr in particular. Perhaps onValueUpdate? `triggered when the input value is updated with a new date string`

**Comment by rwwagner90** [comment]
You can use `flatpickrRef` which let's you get access to the flatpickr instance.

**Comment by kmccullough** [comment]
It appears that the `onValueUpdate` hook doesn't do what I initially thought it did, and is only triggered when the calendar updates the input. It seems the enter key functionality is hard coded into flatpickr. 

I have accessed the instance through `@getFlatpickrRef={{action onFlatpickrInstance}}` and added the appropriate addEventListener to `flatpickrRef.input` to `flatpickrRef.setDate()` on blur.

Thanks for your help.

**Comment by kmccullough** [comment]
I'm sorry if this reopens the ticket. I just wanted to note an issue I had to deal with, for the benefit of anyone who may find this.

I had an error occurring in regard to `ref.config` being `undefined` when I would `ref.setDate()` on `blur`, after the user clicked a date in the calendar. I solved this first by hooking into `ref.config.onChange.push()` to set a boolean `isChanged` to `true`, and then in my `blur` handler resetting that boolean to `false` and deferring the `ref.setDate()` with a `setTimeout()`, to allow for the `onChange` event to occur, and then conditionally doing a `ref.setDate()` if `(!isChanged)` within the `setTimeout()`.
